File: Capstone-Project-dev/Capstone-Project-dev/fe/mobile/ios/ReactNative/Modules/Permission/PermissionBridgeModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(PermissionModule, NSObject)

RCT_EXTERN_METHOD(canAccessFullFeatures:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(canAccessModule:(NSString *)moduleName
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: Capstone-Project-dev/Capstone-Project-dev/fe/mobile/ios/ReactNative/Modules/Permission/PermissionBridgeModule.swift
import Foundation
import React

@objc(PermissionModule)
class PermissionModule: NSObject, RCTBridgeModule {
    static func moduleName() -> String! {
        return "PermissionModule"
    }

    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    private let permissionManager = PermissionManager.shared

    @objc(canAccessFullFeatures:rejecter:)
    func canAccessFullFeatures(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        resolve(permissionManager.canAccessFullFeatures())
    }

    @objc(canAccessModule:resolver:rejecter:)
    func canAccessModule(_ moduleName: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        resolve(permissionManager.canAccessModule(moduleName: moduleName))
    }
}
